src/services/book.service.ts

```diff
@@ -53,10 +55,19 @@ export class BookService {
     return existingBook;
   }
 
-  async sortBook(params: SortBookParams) {
-    const allBook = await repositories.books.sort(params);
+  async list(params: SortBookParams, params2: FilterBookParams, req: any) {
+    const page = serializePagination(req).page;
+    const limit = serializePagination(req).limit;
+    const startIndex = (page - 1) * limit;
+    const count = await repositories.books.count(params2);
+    const allBook = await repositories.books.list({
+      where: params2,
+      orderBy: params,
+      skip: startIndex,
+      take: limit,
+    });
 
-    return allBook;
+    return { allBook, count };
```

Comment:
Please return this shape instead

```
{
  items: Array<T>,
  pagination
}
```

Where the pagination property is an instance of a Pagination class that contains these properties:
- page
- limit
- totalPages
- totalResults